domain/schema/controller.go

```diff
@@ -482,3 +485,65 @@ BEGIN
     VALUES (4, 5, OLD.key, DATETIME('now'));
 END;`)
 }
+
+func upgradeInfoSchema() database.Delta {
+	return database.MakeDelta(`
+CREATE TABLE upgrade_info (
+    uuid             TEXT PRIMARY KEY,
+    previous_version TEXT NOT NULL,
+    target_version   TEXT NOT NULL,
+    created_at       TIMESTAMP NOT NULL,
```

Comment:
what does `created_at` represent? should always be equal to `started_at` right?

Reply:
Perhaps an entry `completed_at` and a boolean comparison against null would be better